app/src/main/java/com/kevinlu/watstats/MainActivity.java

```diff
@@ -1,13 +1,164 @@
 package com.kevinlu.watstats;
 
-import android.os.Bundle;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.viewpager.widget.ViewPager;
+
+import android.os.Bundle;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.francochen.watcard.WatCardAPI;
+import com.francochen.watcard.WatCardService;
+import com.francochen.watcard.model.RequestVerificationToken;
+import com.francochen.watcard.model.balance.BalanceType;
+import com.francochen.watcard.model.balance.Balances;
+import com.kevinlu.watstats.models.AccountBalance;
+import com.kevinlu.watstats.models.MonthlySpending;
+
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class MainActivity extends AppCompatActivity {
 
+    private TextView totalBalance;
+    private ViewPager accountBalances;
+    private ViewPager monthlySpendings;
+    private AccountBalanceAdapter accountBalanceAdapter;
+    private MonthlySpendAdapter monthlySpendAdapter;
+    private List<AccountBalance> accountBalanceList;
+    private List<MonthlySpending> monthlySpendingList;
+    private WatCardAPI api = new WatCardAPI();
+    private WatCardService service = api.createService();
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        totalBalance = findViewById(R.id.total_balance);
+
+        accountBalanceList = new ArrayList<>();
+
+        accountBalanceAdapter = new AccountBalanceAdapter(accountBalanceList, this);
+
+        accountBalances = findViewById(R.id.account_balances);
+        accountBalances.setAdapter(accountBalanceAdapter);
+        accountBalances.setPadding(dpToPx(36), dpToPx(36), dpToPx(205), 0);
+
+        accountBalances.addOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener() {
+            @Override
+            public void onPageSelected(int position) {
+                if (position < accountBalanceList.size() - 1) {
+                    return;
+                }
+                accountBalances.setCurrentItem(position - 1, true);
+            }
+        });
+
+        monthlySpendingList = new ArrayList<>();
+        monthlySpendingList.add(new MonthlySpending("December", "$ 254.98"));
```

Comment:
Why we need these hard codings?

Reply:
You can test with them, but it is better to not include them in a PR.

Reply:
I will remove them now, they;re just for testing.